op-chain-ops/genesis/layer_one.go

```diff
@@ -67,7 +68,7 @@ func BuildL1DeveloperGenesis(config *DeployConfig, dump *gstate.Dump, l1Deployme
 			memDB.CreateAccount(address)
 			memDB.SetNonce(address, account.Nonce)
 
-			balance, ok := new(big.Int).SetString(account.Balance, 10)
+			balance, ok := new(big.Int).SetString(strings.TrimPrefix(account.Balance, "0x"), 16)
```

Comment:
Ah I see here. Maybe a solution would be to use the following type from geth: https://github.com/ethereum/go-ethereum/blob/cd0770ea6855a7704059aa7c591d0e83dcb21231/common/math/big.go#L43

It should work with the new serialization as well as the old serialization. A better solution could be to handle the diff in serialization upstream more such that there is no diff here which would mean that we would be sure that no other code hits this bug. I don't think we need to worry about other code so i'm fine with not doing that and having the diff here

Reply:
Also fine with leaving as is but just heads up that types for handling this sort of thing do exist

Reply:
My preference would be to not have this diff here to avoid any bugs from this change, but this function, `BuildL1DeveloperGenesis` is used 5 times in the monorepo, and I verified this change won't affect anything we have now. However, in my mind, it makes more sense for the balances for allocs to be hex, and it seems unusual for them to have been converted to ints before being converted to bigints - seems like there would also be chance for overflows

Reply:
Refactored to:

```go
var balance math.HexOrDecimal256
err := balance.UnmarshalText([]byte(account.Balance))
if err != nil {
	return nil, fmt.Errorf("failed to parse balance for %s: %w", address, err)
}
memDB.AddBalance(address, (*big.Int)(&balance))
```

Reply:
`(*big.Int)(&balance)` - is this bad to do since it's stripping all of the `HexOrDecimal256` specific functions?

Reply:
This is exactly what we want to do, good work